.github/workflows/build.yml

```diff
@@ -485,8 +487,12 @@ jobs:
         uses: dtolnay/rust-toolchain@nightly
         with:
           components: clippy
-      # do not deny warnings, as nightly clippy sometimes has false negatives
-      - run: ./admin/clippy
+      # Do not deny warnings, as nightly clippy sometimes has false negatives.
+      # - Ignore `clippy::disallowed_types` as a workaround since `disallowed_types` configured in `.clippy.toml`
+      #   is only intended for main `rustls` crate.
+      - run: ./admin/clippy -- --allow clippy::disallowed_types # --allow clippy::incompatible_msrv
+      # Check the main crate for all Clippy nightly warnings; do not deny.
+      - run: cargo clippy -p rustls
```

Comment:
I am proposing this update assuming that my proposal in PR #2311 will be rejected for now. As much as I would like to have CI deny warnings from Clippy nightly, I think this needs consensus from the @rustls maintainers and would rather not wait for this.